### PR TITLE
Update WinML Projection to accept sequence of tensors

### PR DESCRIPTION
**Pull request by smk2007**
WinML currently only accepts sequences of maps. New operators added in opset 11 adds new operators that act on sequence<tensor> types.

Fixes:
1) Allow users to bind sequences of TensorFloat (generally Tensor*) as inputs and outputs. They can be bound as IVector/IVectorView <Tensor*>.
2) Allow unbound outputs for models that output sequence<tensor>.
3) Disable validation that ensured that sequences could only include scalar values.
4) Add WinML tests that exercise sequence<tensor> inputs and outputs.

### CODE REVIEW

File: onnxruntime/core/framework/onnxruntime_typeinfo.cc [hunk 1]
```diff
@@ -126,8 +127,20 @@ OrtStatus* OrtTypeInfo::FromOrtValue(const OrtValue& value, OrtTypeInfo** out) {
   }
 
   if (type->IsTensorSequenceType()) {
-    *out = new OrtTypeInfo(ONNX_TYPE_SEQUENCE);
-    return nullptr;
+    OrtTensorTypeAndShapeInfo* info = nullptr;
+    const auto* tensor_data_type = value.Get<onnxruntime::TensorSeq>().DataType();
+    if (tensor_data_type != nullptr) {
```
Comment (walrusmcd) [inline onnxruntime/core/framework/onnxruntime_typeinfo.cc]: >nsor_data_type != null [](start = 10, length = 22)

Return an error if this is nullptr ?  right now the code just falls through.  if this is a sequence isn't it supposed to have a data type?

**Comment by walrusmcd** [comment]
>     return verify(sequenceFirst.ElementDescriptor(), sequenceSecond.ElementDescriptor());

isn't this the part that does what your new comment on line 192 says?

---
Refers to: winml/lib/Api/impl/FeatureCompatibility.h:206 in 036f8a6. [](commit_id = 036f8a6e5954ce88ca5bdbfc7693899276146d69, deletion_comment = False)

File: onnxruntime/core/framework/onnxruntime_typeinfo.cc (hunk 1 above)
Comment (smk2007) [reply onnxruntime/core/framework/onnxruntime_typeinfo.cc]: Added 
    else {
      return OrtApis::CreateStatus(ORT_FAIL, "OrtValue is TensorSequence type but has no element Tensor DataType associated.")
    }

**Comment by walrusmcd** [comment]
>     return verify(sequenceFirst.ElementDescriptor(), sequenceSecond.ElementDescriptor());

sheil answered offline.   verify() is called for all sort of things, even recursively as it unpeels maps and sequences into their elements.  since seuqences can now support tensors (vs. just scalars) the code @ line 212 had to change to loosen the resitrctions and allow any sort of tensor.

---
In reply to: [602117755](https://github.com/microsoft/onnxruntime/pull/3287#issuecomment-602117755) [](ancestors = 602117755)

---
Refers to: winml/lib/Api/impl/FeatureCompatibility.h:206 in 036f8a6. [](commit_id = 036f8a6e5954ce88ca5bdbfc7693899276146d69, deletion_comment = False)

File: onnxruntime/core/framework/onnxruntime_sequence_type_info.h [hunk 2]
```diff
@@ -12,14 +12,15 @@ class TypeProto;
 
 struct OrtSequenceTypeInfo {
  public:
+  OrtSequenceTypeInfo(OrtTypeInfo* sequence_key_type) noexcept;
```
Comment (pranavsharma) [inline onnxruntime/core/framework/onnxruntime_sequence_type_info.h]: explicit

Comment (smk2007) [reply onnxruntime/core/framework/onnxruntime_sequence_type_info.h]: Added explicit.

**Pull request merged**